src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentsService.java

```diff
@@ -129,4 +153,41 @@ private Map<String, String> makeLocaleMap(final SortedSet<String> localesToQuery
 
         return map;
     }
+
+    private void getFallbackLocaleMessages(Iterator<Locale> fallbackLocalesIter, final Set<String> componentsToFallback,
+                                                                       final Set<String> localesToFallback, final Map<String, Map<String, Map<String, String>>> dataMap) {
+        if (fallbackLocalesIter != null && fallbackLocalesIter.hasNext()) {
+            Locale fallbackLocale = fallbackLocalesIter.next();
+            if (fallbackLocale.toLanguageTag().equals(ConstantsKeys.SOURCE)) {
+                fallbackLocale = Locale.forLanguageTag(ConstantsKeys.LATEST);
+            }
+            Set<Locale> localeSet = new HashSet<Locale>();
+            localeSet.add(fallbackLocale);
+            Map<Locale, Map<String, Map<String, String>>> localeDataMap = getTranslation(componentsToFallback, localeSet, fallbackLocalesIter);
+            Map<String, Map<String, String>> componentsMap = localeDataMap.get(fallbackLocale);
+            if (componentsMap != null && !componentsMap.isEmpty()) {
+                final SortedSet<String> localesToFallback2 = new TreeSet<>();
+                final SortedSet<String> componentsToFallback2 = new TreeSet<>();
+
+                final Iterator<?> localeIter = localesToFallback.iterator();
+                while (localeIter.hasNext()) {
+                    String locale = (String) localeIter.next();
+                    for (String component : componentsToFallback) {
+                        Map<String, String> componentMap = componentsMap.get(component);
+                        if (componentMap != null) {
+                            if (dataMap.get(locale).get(component) == null)
+                                dataMap.get(locale).put(component, componentMap);
+                        } else {
+                            localesToFallback2.add(locale);
+                            componentsToFallback2.add(component);
+                        }
+                    }
+                }
+                if(!componentsToFallback2.isEmpty())
+                    throw new VIPJavaClientException(FormatUtils.format(ConstantsMsg.GET_MESSAGES_FAILED_ALL, componentsToFallback2.toString(), localesToFallback2.toString()));
+            } else {
+                getFallbackLocaleMessages(fallbackLocalesIter, componentsToFallback, localesToFallback, dataMap);
+            }
+        }
+    }
```

Comment:
Throw exception here?
```suggestion
                if(!componentsToFallback.isEmpty())
                    throw new VIPJavaClientException(FormatUtils.format(ConstantsMsg.GET_MESSAGES_FAILED_ALL, componentsToFallback.toString(), localesToFallback.toString()));

    }
```